### PR TITLE
feat(core) allow the NestModule.configure() method to be async

### PR DESCRIPTION
**Pull request by zailic**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1384

## What is the new behavior?
Allows the NestModule.configure() method to be asynchronous. More details can be found on #1384.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: packages/common/interfaces/modules/nest-module.interface.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { MiddlewareConsumer } from '../middleware/middleware-consumer.interface';
 
 export interface NestModule {
-  configure(consumer: MiddlewareConsumer): MiddlewareConsumer | void;
+  configure(consumer: MiddlewareConsumer): Promise<MiddlewareConsumer | void>;
```
Comment (kamilmysliwiec) [inline packages/common/interfaces/modules/nest-module.interface.ts]: I think that we can just remove the return type here entirely.

Comment (zailic) [reply packages/common/interfaces/modules/nest-module.interface.ts]: I think you're absolutely right.

**Comment by kamilmysliwiec** [comment]
Could you rebase your PR to the latest changes on master? It seems that it contains lots of renovate commits now:)

**Comment by zailic** [comment]
ups... sorry about that! I'll do it right now.

**Comment by zailic** [comment]
The header of this issue is a lie:). I don't want to merge 54 commits... Bellow are the changes for CR:
https://github.com/nestjs/nest/compare/master...zailic:IonutZailic-IssueNo1384

**Comment by kamilmysliwiec** [comment]
Thanks!

**Comment by zailic** [comment]
Thank you! Can't wait for 6.6.0 release:)

**Pull request merged**